include/math_utils.h

```diff
@@ -127,6 +127,33 @@ constexpr T1 left_shift_signed(T1 value, T2 amount)
 	return static_cast<T1>(shifted);
 }
 
+template <typename T>
+int8_t clamp_to_int8(const T val)
+{
+	const auto tmp = std::clamp(val,
+	                            static_cast<T>(INT8_MIN),
+	                            static_cast<T>(INT8_MAX));
+	return static_cast<int8_t>(tmp);
+}
+
+template <typename T>
+int16_t clamp_to_int16(const T val)
+{
+	const auto tmp = std::clamp(val,
+	                            static_cast<T>(INT16_MIN),
+	                            static_cast<T>(INT16_MAX));
+	return static_cast<int16_t>(tmp);
+}
+
+template <typename T>
+int32_t clamp_to_int32(const T val)
+{
+	const auto tmp = std::clamp(val,
+	                            static_cast<T>(INT32_MIN),
+	                            static_cast<T>(INT32_MAX));
+	return static_cast<int32_t>(tmp);
+}
+
```

Comment:
I've added unit tests for these (on a branch), if you can cherry-pick them in: 2434cc89d32

Reply:
And unsigned handling to `clamp_to` in 66091cfd7f0f90259552c639dbfda912e69be3de

Reply:
OK, cherry-picked - not closing the conversation, as I need to review the code if these templates can be reused elsewhere in the mouse code now.